tests/jballerina-unit-test/src/test/resources/test-src/query/query_with_closures.bal

```diff
@@ -172,6 +172,116 @@ function testClosureInQueryActionInDo2() {
+        self.i += 2;
+        if (self.i >= 10) {
+            return error("Error");
+        }
+        return {value: self.i};
+    }
+}
+
+string str1 = "string-1";
+
+function testClosureInQueryActionInDo3() returns error? {
+    string str2 = "string-2";
+    EvenNumberGenerator evenGen = new;
+    stream<int, error?> evenNumberStream = new (evenGen);
+    int|string str3 = "string-3";
+
+    if (str3 is int) {
+    } else {
+        check from var _ in evenNumberStream
+        do {
+            string _ = str1;
+            string _ = str2;
+            string _ = str3;
+            int length1 = str1.length();
+            int length2 = str2.length();
+            int length3 = str3.length();
+
+            assertEquality(str1, "string-1");
+            assertEquality(str2, "string-2");
+            assertEquality(str3, "string-3");
+            assertEquality(length1, 8);
+            assertEquality(length2, 8);
+            assertEquality(length3, 8);
+        };
+  }
+}
+
+class A {
+    public string? name;
+
+    public function init(string? name) {
+        self.name = name;
+    }
+
+    public function getName() returns string|error {
+        if self.name is () {
+            return error("Null value found for name attribute");
+        }
+        return <string> self.name;
+    }
+}
+
+A object1 = new ("John");
+
+function testClosureInQueryActionInDo4() returns error? {
+    A object2 = new ("Jane");
+    EvenNumberGenerator evenGen = new ();
+    stream<int, error?> evenNumberStream = new (evenGen);
+    A|error object3 = new ("Anne");
+
+    if (object3 is error) {
+    } else {
+        check from var _ in evenNumberStream
+        do {
+            A _ = object1;
+            A _ = object2;
+            A _ = object3;
+
+            string objectName1 = check object1.getName();
+            string objectName2 = check object2.getName();
+            string objectName3 = check object3.getName();
+
+            assertEquality("John", objectName1);
+            assertEquality("Jane", objectName2);
+            assertEquality("Anne", objectName3);
+        };
+  }
+}
+
+function testClosureInQueryActionInDo5() returns error? {
+    A object2 = new ("Jane");
+    EvenNumberGenerator evenGen = new ();
+    stream<int, error?> evenNumberStream = new (evenGen);
+    A|error object3 = new ("Anne");
+
+    if (object3 is error) {
+    } else {
+    check from var _ in evenNumberStream
+    do {
+        check from var _ in evenNumberStream
+        do {
+            A _ = object1;
+            A _ = object2;
+            A _ = object3;
+
+            string objectName1 = check object1.getName();
+            string objectName2 = check object2.getName();
+            string objectName3 = check object3.getName();
+
+            assertEquality("John", objectName1);
+            assertEquality("Jane", objectName2);
+            assertEquality("Anne", objectName3);
+        };
+    };
+  }
+}
+
```

Comment:
Can you add some tests for query expressions, nested query actions and expressions

Reply:
will add.

Reply:
added